scripts/create_deprecated_modules.lean

```diff
@@ -0,0 +1,371 @@
+  return substring.toString
+
+/--
+`getHeaderFromFileName fname keepTrailing` is similar to `getHeader`, except that it assumes that
+`fname` is the actual path of a file, and uses `fname`'s content as input to `getHeader`.
+-/
+def getHeaderFromFileName (fname : String) (keepTrailing : Bool) : IO String := do
+  getHeader fname (← IO.FS.readFile fname) keepTrailing
+
+/--
+`mkDeprecationWithDate date customMessage` returns the formatted syntax
+`deprecated_module "customMessage" (since := "date")`,
+where the date is of the form `YYYY-MM-DD`.
+-/
+def mkDeprecationWithDate (date : String)
+    (customMessage : Option String := some "Auto-generated deprecation") :
+    CommandElabM Format := do
+  let msgStx := customMessage.map Syntax.mkStrLit
+  let dateStx := Syntax.mkStrLit date
+  let stx ← `(command|deprecated_module $[$msgStx]? (since := $dateStx))
+  liftCoreM <| PrettyPrinter.ppCategory `command stx
+
+/--
+`mkDeprecation customMessage` returns the formatted syntax
+`deprecated_module "customMessage" (since := "YYYY-MM-DD")`, where the date is today's date.
+-/
+def mkDeprecation (customMessage : Option String := some "Auto-generated deprecation") :
+    CommandElabM Format := do
+  -- Get the current date in UTC: we don't want this to depend on the user computer's time zone.
+  let date := s!"{(← Std.Time.DateTime.now (tz := .UTC)).toPlainDate}"
+  mkDeprecationWithDate date customMessage
+
+/--
+The command `#create_deprecated_module filePath (<comment>)? (rename_to <fname>) (write)?`
+generates a module deprecation.
+
+Writing
+```lean
+#create_deprecated_module path/To/DeletedFile.lean "This file is no longer relevant" rename_to "Mathlib/Path/To/Rename.lean"
+```
+checks that `path/To/DeletedFile.lean` is not currently present, but was present in `Mathlib/`
+at some point.
+
+If the check is successful, then it reports on its findings, shows how the corresponding
+deprecated module should look like, using `"This file is no longer relevant"` as the (optional)
+<comment>.
+
+If the message is not explicitly used, `#create_deprecated_module` defaults to
+`"Auto-generated deprecation"`.
+If you wish there to be no comment, use `#create_deprecated_module path/To/DeletedFile.lean ""`.
+
+If `rename_to "Mathlib/Path/To/Rename.lean"` is present, then instead of copying over the imports
+from a deleted file, it uses `import Mathlib.Path.To.Rename`.
+
+Finally, if everything looks correct, adding a final `write` actually generates the file:
+```lean
+#create_deprecated_module path/To/DeletedFile.lean "This file is no longer relevant" write
+```
+-/
+syntax "#create_deprecated_module " str (ppSpace str)? (&" rename_to " str)? (&" write")? ppLine :
+  command
+
+/-- `processPrettyOneLine log msg` takes as input two strings `log` and `msg`.
+It expects `log` to be a line in the output of `git log --pretty=oneline`:
+it should look like `<hash> <PRdescr>`.
+
+It also expects `msg` to be either `last modified` or `deleted` and
```

Comment:
Let's rephrase this: this function doesn't inspects `msgs`... so how about something like this?
```suggestion
It returns the pair `(<hash>, <msg> in <PRdescr> <hash> <diff of file wrt previous commit>)`,
formatted as a collapsible message. In practice, `msg` is either `last modified` or `deleted`.